qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -418,6 +419,42 @@ def name(self) -> str:
         """
         return self._name
 
+    def update_name(self, name: str) -> str:
+        """Update the name of this job set.
+
+        Args:
+            name: The new `name` for this job set.
+
+        Returns:
+            The new name associated with this job set.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the input job name is not a string.
+            IBMQJobManagerUpdateError: If there was an error with updating the names
+                for the jobs in the job set.
+        """
+        if not isinstance(name, str):
+            raise IBMQJobManagerInvalidStateError(
+                '"{}" of type "{}" is not a valid job name. '
+                'The job name needs to be a string.'.format(name, type(str)))
+
+        for i, job in enumerate(self.jobs()):
+            if job:
+                try:
+                    _ = job.update_name("{}_{}_".format(name, i))
+                except IBMQJobUpdateError as ex:
+                    raise IBMQJobManagerUpdateError(
+                        "An error occurred when updating the name for this job set: {}"
+                        "As a result, some of the jobs in this set may not have been "
+                        "updated. Please try updating the job set's name again "
```

Comment:
Don't tell the user to "try again" unless you know it's a temporary networking error (which everyone knows happens from time to time) because it makes the code looks buggy and the user would wonder why you don't just retry in the code.